elasticdl/pkg/common/tensor.go

```diff
@@ -16,21 +16,50 @@ type Tensor struct {
 	Indices []int64
 }
 
-func getDimProduct(dim []int64) int64 {
+// NewTensor create a new tensor with name and dim
+func NewTensor(name string, dim []int64) *Tensor {
+	var t Tensor
+	t.Name = name
+	t.Value = make([]float32, GetDimProduct(dim))
+	t.Dim = dim
+	return &t
+}
+
+// NewEmbeddingVector create a new tensor with dim
+func NewEmbeddingVector(dim int64) *Tensor {
+	var t Tensor
+	t.Value = make([]float32, dim)
+	t.Dim = []int64{1, dim}
+	return &t
+}
+
+// GetDimProduct get the number of the elements of a tensor of this dim
+func GetDimProduct(dim []int64) int64 {
 	var size int64 = 1
 	for _, d := range dim {
 		size *= d
 	}
 	return size
 }
 
+// Subtensor get the part reference of the tensor
+func (t *Tensor) Subtensor(begin int64, len int64) *Tensor {
```

Comment:
It seems that this function contains two side effects:

- takes a slice of a tensor, and returns a subtensor
- resizes a tensor, changes the dim

We could split this function into two functions to make it more clear and general:

```go
func (t *Tensor) Resize(dim int64[]) {
    t.dim = dim;
}

// Slice returns a slice of a Tensor, the slice could only be done at dim[0]
func (t *Tensor) Slice(begin int64, end int64) *Tensor {
    ...
    copy(slice.Dim, t.Dim);
    slice.Dim[0] = end - begin;
    ....
}
```

Reply:
this function does not copy a tensor, it returns a reference to a specific part. it does not resize the tensor, the dim of a tensor has to be an array, that's why I use {1, dim} to fill the field.

Reply:
or I could just use {dim} to fill the field

Reply:
wrapped by a function `common.Tensor.AtRow(int64)`

Reply:
Got it. Then we could rename `Subtensor` to `subTensor` to make it a internal function.